### PR TITLE
Upgrade React to 15.5.4

### PR DESCRIPTION
**Pull request by oshalygin**
Greetings gents, this PR upgrades us to the latest version of React, `15.5.4`.  Most of the codebase already supports this migration so there's really no code changes other than package updates.

- This change is relatively straightforward with the
  current codebase as nearly all of the major changes
  are already preemptively addressed, namely not using
  React.createClass, react-test-utils, and accessing
  PropTypes from the main React object.
- Enzyme was updated to the latest patch version along
  with installation of react-test-renderer.  The
  react-addons-test-utils package is being deprecated
  with React >=15.5.  More information regarding enzyme
  installation instructions with React >=15.5:
https://github.com/airbnb/enzyme\#installation

- react-redux was updated to the latest patch version
  which was updated to reflect usage of PropTypes from
  the 'prop-types' package instead of the main React
  package.
- react-router was updated to the latest patch version
  under v3.X to reflect the usage of PropTypes from the
  'prop-types' package instead of the main React package.
- Removal of wildcard flags from certain packages in
  the package manifest.
- Removal of react-addons-test-utils from the list of
  dependencies as it is no longer used.

More information in regards to the release notes of
React v15:
https://facebook.github.io/react/blog/2017/04/07/react-v15.5.0.html

### CODE REVIEW

**Comment by oshalygin** [comment]
No warnings in the console or the application:

image

image

File: package.json [hunk 1]
```diff
@@ -59,26 +59,26 @@
     "connect-history-api-fallback": "1.3.0",
     "coveralls": "2.11.15",
     "css-loader": "0.26.1",
-    "enzyme": "2.6.0",
+    "enzyme": "2.8.2",
     "eslint": "3.12.2",
     "eslint-plugin-import": "2.2.0",
     "eslint-plugin-react": "6.8.0",
     "eslint-watch": "2.1.14",
-    "extract-text-webpack-plugin": "^2.1.0",
```
Comment (nickytonline) [inline package.json]: Not sure how carrots got in here. Thanks for putting those to fixed versions.

**Comment by nickytonline** [comment]
@oshalygin, thanks for taking the time with this PR. The PR is well crafted. 💯 Can you create an issue and link it to the PR? It just makes it easier to track what's been going into the codebase. Cheers.

**Comment by coryhouse** [comment]
Agreed, looks great to me! 👍

**Comment by oshalygin** [comment]
Thanks gents!

**Comment by kwelch** [comment]
Do we not use `prop-types` in the demo app at all?

**Comment by coryhouse** [comment]
@kwelch prop-types is referenced in the demo app. Why ya ask?

**Comment by kwelch** [comment]
I didn't see in the packages added like but I know they moved it out to a separate package. We may have caught that change already.

**Comment by oshalygin** [comment]
@kwelch yeah that migration happened in this PR:
https://github.com/coryhouse/react-slingshot/pull/418

**Comment by kwelch** [comment]
I thought so but could not recall off the top of my head. Thanks!

**Pull request merged**